WordPress/src/main/java/org/wordpress/android/ui/plans/models/Plan.java

```diff
@@ -0,0 +1,218 @@
+    private int mStore;
+    private int mMulti;
+    private String mSupportDocument;
+    private String mCapability;
+    private List<Integer> mBundleProductIds;
+    private List<String> mFeatures;
+
+    public Plan(JSONObject planJSONObject) throws JSONException {
+        mProductID = planJSONObject.getLong("product_id");
+        mProductName = planJSONObject.getString("product_name");
+        mProductNameEnglish = planJSONObject.getString("product_name_en");
+
+        // Unfold prices object
+        JSONObject priceJSONObject = planJSONObject.getJSONObject("prices");
+        JSONArray priceKeys = priceJSONObject.names();
+        if (priceKeys != null) {
+            for (int i=0; i < priceKeys.length(); i ++) {
+                String currentKey = priceKeys.getString(i);
+                int currentPrice = priceJSONObject.getInt(currentKey);
+                mPrices.put(currentKey, currentPrice);
+            }
+        }
+
+        mProductNameShort = planJSONObject.getString("product_name_short");
+        mProductSlug = planJSONObject.getString("product_slug");
+        mTagline = planJSONObject.getString("tagline");
+        mShortdesc = planJSONObject.getString("shortdesc");
+        mDescription = planJSONObject.getString("description");
+        mCost = planJSONObject.getLong("cost");
+        mBillPeriod = planJSONObject.getInt("bill_period");
+        mProductType = planJSONObject.getString("product_type");
+        isAvailable = JSONUtils.isStringTrue(planJSONObject, "available");
+        mBillPeriodLabel = planJSONObject.getString("bill_period_label");
+        mPrice = planJSONObject.getString("price");
+        mFormattedPrice = planJSONObject.getString("formatted_price");
+        mRawPrice = planJSONObject.getInt("raw_price");
+
+        // Optionals
+        mWidth = planJSONObject.optInt("width");
+        mHeight = planJSONObject.optInt("height");
+        mSaving = planJSONObject.optLong("saving", 0L);
+        mOriginal = planJSONObject.optLong("original", mCost);
+        mFormattedOriginalPrice = planJSONObject.optString("formatted_original_price");
+        mSupportDocument = planJSONObject.optString("support_document");
+        mCapability = planJSONObject.optString("capability");
+        mStore = planJSONObject.optInt("store");
+        mMulti = planJSONObject.optInt("multi");
+
+        if (planJSONObject.has("bundle_product_ids")) {
+            JSONArray bundleIDS = planJSONObject.getJSONArray("bundle_product_ids");
+            mBundleProductIds = new ArrayList<>(bundleIDS.length());
+            for (int i=0; i < bundleIDS.length(); i ++) {
+                int currentBundleID = bundleIDS.getInt(i);
+                mBundleProductIds.add(currentBundleID);
+            }
+        }
+
+        if (planJSONObject.has("feature_1")) {
+            // At least one feature is available
+            mFeatures = new ArrayList<>();
+            for (int i=1; i < 50; i ++) {
```

Comment:
Are the features always sequential (ie: no numbers skipped)? If so, a `while` loop which keeps reading features until none exist would work. Off the top of my head:
```
int featureNumber = 1;
while (planJSONObject.has("feature_" + featureNumber)) {
    mFeatures.add(planJSONObject.getString("feature_" + featureNumber));
    featureNumber++;
}
```